src/os/bluestore/BlueFS.h

```diff
@@ -38,8 +38,33 @@ enum {
   l_bluefs_last,
 };
 
-class BlueFS {
+class BlueFS : public md_config_obs_t {
 public:
+  // config observer
+  const char** get_tracked_conf_keys() const override {
+    static const char* KEYS[] = {
+      "bluefs_alloc_size",
+      "bluefs_max_prefetch",
+      "bluefs_min_log_runway",
+      "bluefs_max_log_runway",
+      "bluefs_log_compact_min_ratio",
+      "bluefs_log_compact_min_size",
+      "bluefs_min_flush_size",
+      "bluefs_compact_log_sync",
+      "bluefs_buffered_io",
+      "bluefs_sync_write",
+      "bluefs_allocator",
+      "bluefs_preextend_wal_files",
+      NULL
+    };
+    return KEYS;
+  }
+
+  void handle_conf_change(const struct md_config_t* conf,
+    const std::set<std::string>& changed) override {
+      // do some stuff here...
```

Comment:
looks like they are already reserved, they just need to be copied into local member fields here, so that we don't call get_val<> in the hot paths.